### PR TITLE
Hibernate

### PR DESCRIPTION
**Pull request by Panic7**

### CODE REVIEW

File: src/main/java/com/example/demo/DemoApplication.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    private final ConsoleInOut console;
+
+    public static void main(String[] args) {
+        SpringApplication.run(DemoApplication.class, args);
+    }
+
+    @Override
+    public void run(String... args) {
+        console.getTournament().getTeamService().saveAll(List.of(
```
Comment (vova-pereb) [inline src/main/java/com/example/demo/DemoApplication.java]: This breaks DI.
Beans should never be accessed by getters.

File: src/main/java/com/example/demo/model/Match.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+@ToString(of = "id")
+@Table(name = "matches")
+@FieldDefaults(level = AccessLevel.PRIVATE)
+public class Match {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(unique = true, nullable = false)
+    Integer id;
+
+    @ManyToOne(fetch = FetchType.EAGER, optional = false)
```
Comment (vova-pereb) [inline src/main/java/com/example/demo/model/Match.java]: Eager fetch type is forbidden.

File: src/main/java/com/example/demo/service/base/GeneralService.java [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+import javax.persistence.EntityNotFoundException;
+import java.util.List;
+
+@RequiredArgsConstructor
+@FieldDefaults(level = AccessLevel.PRIVATE, makeFinal = true)
+public class GeneralService<ENTITY, ID extends Integer> {
+    CrudRepository<ENTITY, ID> repository;
+
+    public ENTITY getById(ID id) {
```
Comment (vova-pereb) [inline src/main/java/com/example/demo/service/base/GeneralService.java:19]: `org.springframework.transaction.annotation.Transactional`
Should be used if you access DB.

File: src/test/java/com/example/demo/DemoApplicationTests.java [hunk 4]
```diff
@@ -0,0 +1,13 @@
+package com.example.demo;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@SpringBootTest
+class DemoApplicationTests {
```
Comment (vova-pereb) [inline src/test/java/com/example/demo/DemoApplicationTests.java]: Redundant

**Review by vova-pereb — CHANGES_REQUESTED** [review]

**Comment by Panic7** [comment]
@vova-pereb

**Pull request closed without merging**